### DISCUSSION TITLE
Typescript

### DISCUSSION THREAD

**Discussion by ShakeelKhalid1913** [expo/router#262] [General]
how can i use typescript in expo-router project?

**Comment by ShaneKeney** [comment]
Quickest way to get started that I know of would be the following:
I'll assume your using yarn.

```
npx create-react-native-app -t with-router
```

Add the following dependencies:

```
yarn add -D typescript @types/react @types/react-native @types/react-dom
```

Run `yarn install`
Run `yarn start`

Open templated Expo Router app and it will automatically create your `app/index.js` file upon clicking button on open.

You can then change this to `index.tsx` and everything should be working with TS.

**Comment by arnaudbzn** [comment]
This command works for me
```
npx create-react-native-app -t with-typescript --with-router
```

**Comment by EvanBacon** [comment]
Convert any file to TS while `npx expo start` is running and TypeScript will be setup.
